app/models/session.rb

```diff
@@ -125,10 +115,28 @@ def self.from_location(data)
     where(:streams => {:id => streams_ids.uniq})
   end
 
+  def self.filter_by_time_range(sessions, time_from, time_to)
+    time_from = Time.strptime(time_from.to_s, '%s')
+    time_to = Time.strptime(time_to.to_s, '%s')
+
+    sessions.where(
+      "(start_time_local BETWEEN :time_from AND :time_to)
+      OR
+      (end_time_local BETWEEN :time_from AND :time_to)
+      OR
+      (:time_from BETWEEN start_time_local AND end_time_local)",
+      :time_from => time_from, :time_to => time_to)
+      .local_minutes_range(minutes_of_day(time_from), minutes_of_day(time_to))
+  end
+
   def self.whole_day?(time_from, time_to)
     time_from == FIRST_MINUTE_OF_DAY && time_to == LAST_MINUTE_OF_DAY
   end
 
+  def self.minutes_of_day(time)
```

Comment:
Have you considered having this in another module? It seems like it's not related to Session